CompilerSource/ipc/Shader.fbs

```diff
@@ -0,0 +1,15 @@
+
+enum ShaderType : byte { GLSL, GLSLES, HLSL9, HLSL11 }

```

Comment:
Is this extensible? I know we are on Vulkan now, but what if we later want to add a new HLSL version in between HLSL and Vulkan at the end. This is an area where we will likely want to update somewhat frequently.

Enums are discussed here:
https://google.github.io/flatbuffers/flatbuffers_guide_writing_schema.html